controllers/discovery/searchTopic.js

```diff
@@ -14,12 +14,6 @@ const Search = async (req, res) => {
   const {userId} = req;
   const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
 
-  if (q.length < 2)
-    return res.status(200).json({
-      success: true,
-      message: 'Your search characters is too few, please input 3 or more characters for search'
-    });
-
   try {
     const topics = await sequelize.query(
       `
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Removing the character length check for search queries might lead to performance issues. If a user enters a single character or an empty string, it could potentially return a large number of results, which could slow down the system. Consider adding a reasonable minimum length check for the search query.
```diff
+  if (q.length < 2)
+    return res.status(200).json({
+      success: true,
+      message: 'Your search characters is too few, please input 3 or more characters for search'
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->